### ISSUE TITLE
Firefox and CSS grid and cover block

### ISSUE THREAD

**Issue by johngodley** [Automattic/block-experiments#27]
There's a difference between how Chrome/Safari and Firefox displays a cover block inside a grid block. 

This is based on an issue reported on [this page](https://amwpapptesting.wordpress.com/tiled-gallery-and-other-testing/) where the bottom black box leaks outside the post in Firefox. I think I've narrowed the complicated page down to this codepen:

https://codepen.io/johngodley/pen/vYOYxJe

In Safari/Chrome the black box is about the same height as the yellow one. In Firefox the black box is a lot bigger. 

The issue seems to be caused by `wp-block-cover` having a `height: 100%` when it follows something else that also has height. That seems to stretch the cover, and grid. Without that the height drops to the `min-height: 430px`.

Is there anything we can do to the grid block to prevent this?

**Comment by johngodley** [comment]
Sorry @jasmussen, throwing this over to you for ideas! Noting that this is low priority

**Comment by jasmussen** [comment]
Just took a quick look, curious indeed.

```
.wp-block-jetpack-layout-grid-column {
overflow: hidden;
}
```

That fixes it in a "brute force" way, but I'm still unsure what the issue is in the first place. Your detective work helps. 

What happens, it seems, is that the 100% height block grows the 430px min-height, plus the same amount of pixels as the preceeding block. So for example if the yellow block is 48px tall, then the cover block is 478px tall. 

This is another fix, 

```
.wp-block-jetpack-layout-grid-column {
display: flex;
flex-direction: column;
}
```

But I'm also not sure why that works. Probably because it's the _grid_ display that makes the succeeding cover block be too tall, and changing that to flex on the child block fixes that. But that might also break the grid block entirely, I'm not sure. What do you think?

**Comment by johngodley** [comment]
Hmm, yeah, they seem trouble causing!

I've tried searching for known Firefox bugs but couldn't turn anything up.

I'm sure the `height: 100%` is needed for the cover block, but would it cause other problems to disable that when inside the grid block on Firefox?

Does it even make sense to have a cover block inside a grid? I'm trying to think of a situation where it might be useful to do so. The contents of the original post seem to be a mess of groups, containers, and grids, with some purpose in mind. The grid isn't even used as a grid but more of misguided container. The same design could be achieved by just taking the cover block out of the grid.

All that to say, as it's possible that this could get fixed in Firefox, and that it's maybe an unlikely setup, is it reasonable to call this a known issue with the workaround being to just move the cover block out of the grid?

**Comment by jasmussen** [comment]
> I'm sure the height: 100% is needed for the cover block, but would it cause other problems to disable that when inside the grid block on Firefox?

I don't know that this solution will scale, as other blocks will likely leverage this type of height too.  Though yes, it is a bit weird that the block has this height. The min-height should be sufficient, combined with the ability to vertically size it. 

So, I just found a "fix". This fix appears to be harmless for the time being, but it's obviously worth testing in a variety of browsers.

If the cover image, or _the block that applies a 100% height_, receives a max-height of 0, it works in Firefox. I don't know if this is the equivalent of unsetting the 100% height, it probably is? But it seems worth trying.

In other words:

```
.wp-block-cover {
max-height: 0;
}
```

works. 

If there are side-effects, we might actually consider reporting the 100% height as an upstream bug.
